tests/expectations/metrics/test_metrics_util.py

```diff
@@ -272,3 +295,149 @@ def test_get_unexpected_indices_for_multiple_pandas_named_indices_named_unexpect
+            "TRAVEL_AGENTS",
+        ],
+    ],
+)
+def test_get_dbms_compatible_metric_domain_column_kwargs(
+    sa,
+    column_names_all_uppercase: list[str],
+    input_column_name: str,
+    output_column_name: Union[str, sqlalchemy.quoted_name],
+    confirm_not_equal_column_name: Union[str, sqlalchemy.quoted_name],
+):
+    quoted_column_name: sqlalchemy.quoted_name = sqlalchemy.quoted_name(
+        value="travel_agents", quote=True
+    )
+    test_column_names: list[str] = column_names_all_uppercase + [quoted_column_name]
+
+    metric_domain_kwargs: dict
+
+    metric_domain_kwargs = get_dbms_compatible_metric_domain_kwargs(
+        metric_domain_kwargs={"column": input_column_name},
+        batch_columns_list=test_column_names,
+    )
+    assert metric_domain_kwargs["column"] == output_column_name
+    if confirm_not_equal_column_name:
+        assert metric_domain_kwargs["column"] != confirm_not_equal_column_name
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "input_column_name_a, input_column_name_b, output_column_name_a, output_column_name_b",
+    [
+        [
+            "ARTISTS",
+            sqlalchemy.quoted_name(value="travel_agents", quote=True),
+            "ARTISTS",
+            sqlalchemy.quoted_name(value="travel_agents", quote=True),
+        ],
+        [
+            "ARTISTS",
+            "travel_agents",
+            "ARTISTS",
+            sqlalchemy.quoted_name(value="travel_agents", quote=True),
+        ],
+    ],
+)
+def test_get_dbms_compatible_metric_domain_column_pair_kwargs(
+    sa,
+    column_names_all_uppercase: list[str],
+    input_column_name_a: str,
+    input_column_name_b: str,
+    output_column_name_a: Union[str, sqlalchemy.quoted_name],
+    output_column_name_b: Union[str, sqlalchemy.quoted_name],
+):
+    quoted_column_name: sqlalchemy.quoted_name = sqlalchemy.quoted_name(
+        value="travel_agents", quote=True
+    )
+    test_column_names: list[str] = column_names_all_uppercase + [quoted_column_name]
+
+    metric_domain_kwargs: dict
+
+    metric_domain_kwargs = get_dbms_compatible_metric_domain_kwargs(
+        metric_domain_kwargs={
+            "column_A": input_column_name_a,
+            "column_B": input_column_name_b,
+        },
+        batch_columns_list=test_column_names,
+    )
+    assert metric_domain_kwargs["column_A"] == output_column_name_a
+    assert metric_domain_kwargs["column_B"] == output_column_name_b
+
+
+@pytest.mark.unit
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "input_column_list, output_column_list",
+    [
+        [
+            ["ARTISTS", "travel_agents", "FarmAnimals", "Household_Pets"],
+            [
+                "ARTISTS",
+                sqlalchemy.quoted_name(value="travel_agents", quote=True),
+                sqlalchemy.quoted_name(value="FarmAnimals", quote=True),
+                sqlalchemy.quoted_name(value="Household_Pets", quote=True),
+            ],
```

Comment:
Consider adding an `id` to each of the params that describes what is being tested. It can help with readability.

Reply:
@anthonyburdi Great idea -- in next pull request coming imminently.  Thanks!